app/tests/lib/db_testing.py

```diff
@@ -25,21 +27,28 @@ def create_isolated_db(monkeypatch) -> db.DBClient:
     db_client = db.PostgresDBClient()
     with db_client.get_connection() as conn:
         _create_schema(conn, schema_name)
-        try:
-            yield db_client
-        finally:
+
+    try:
+        yield db_client
+
+    finally:
+        with db_client.get_connection() as conn:
```

Comment:
I agree, but I'm not sure if there's any alternative. We need a connection to run the delete schema command. If this were to fail, we'd just leave that schema behind. Due to me missing something for the delete schema, while I was testing this I ended up with 50 test schemas and it didn't cause any issues, so worst case is not cleaning up properly

Reply:
> 'm not sure if there's any alternative

did the original code not work? in the old code, we indented the try/catch block so that we can use the same connection that created the schema to drop the schema e.g.:

```python
    with db_client.get_connection() as conn:
        _create_schema(conn, schema_name)

        try:
            yield db_client

        finally:
            with db_client.get_connection() as conn:
```

i guess i'm trying to understand why we need to exit the connection context before we do the try/catch

Reply:
I had originally done that in the NJ codebase and ran into errors if the connection failed, but trying again (running a unit test, pause in the middle, deleting a table from the schema so the client fails) still dropped the schema. So either something is slightly different here, or 2.0 fixed whatever problem I was hitting. Either way, I can do that.